appveyor.yml

```diff
@@ -0,0 +1,335 @@
+                    Write-Host "Unable to find $key property" -foregroundcolor 'white' -backgroundcolor 'red'
+                    exit 1
+                }
+                $property.InnerText = $value
+            }
+            $config.Save($configFile)
+        }
+
+    - ps: >-
+        Invoke-Command -ScriptBlock {
+            switch ($env:db) {
+                'Firebird' {
+                    # Install Firebird
+                    # Firebird zip is cached, check if we have it, otherwise download it
+                    If (-Not (Test-Path C:\firebird\Firebird-3.0.2.32703-0_x64.zip)) {
+                        Write-Host "Firebird zip seemingly not cached, fetching it..."
+                        If (-Not (Test-Path C:\firebird)) {
+                            mkdir C:\firebird
+                        }
+                        cd C:\firebird
+                        # Download FireBird zip from Sourceforge. Mirror on free.fr due to my ISP, likely not the best one for AppVeyor VM.
+                        Start-FileDownload 'https://freefr.dl.sourceforge.net/project/firebird/firebird-win64/3.0.2-Release/Firebird-3.0.2.32703-0_x64.zip'
+                    }
+                    Else {
+                        cd C:\firebird
+                    }
+                    7z x Firebird-3.0.2.32703-0_x64.zip
+                    mkdir Data
+                    # Declare nhibernate db
+                    Add-Content -Path '.\databases.conf' -Value "`r`nnhibernate = C:\firebird\Data\nhibernate.fdb"
+                    # Minimal db settings
+                    Add-Content -Path '.\firebird.conf' -Value "`r`nAuthServer = Srp`r`nAuthClient = Srp`r`nUserManager = Srp`r`nWireCrypt = Enabled"
+                    # Create SYSDBA account
+                    New-Item SYSDBA.sql -Type File
+                    Add-Content -Path '.\SYSDBA.sql' -Value "create user SYSDBA password 'masterkey';`r`ncommit;`r`nquit;"
+                    .\isql -user sysdba employee -input SYSDBA.sql
+                    # Start Firebird
+                    .\firebird -a
+                }
+                'MySQL' {
+                    Start-Service 'MySQL57'
+                    # Create nhibernate database (not handled by NHibernate.TestDatabaseSetup.dll)
+                    $env:MYSQL_PWD = 'Password12!'
+                    $cmd = '"C:\Program Files\MySQL\MySQL Server 5.7\bin\mysql" -e "create database nhibernate;" --user=root'
+                    iex "& $cmd"
+                }
+                'Odbc' { Start-Service 'MSSQL$SQL2017' }
+                'PostgreSQL' {
+                    # Enable prepared transactions
+                    Add-Content -Path 'C:\Program Files\PostgreSQL\10\data\postgresql.conf' -Value "`r`nmax_prepared_transactions = 100"
+                    Start-Service 'postgresql-x64-10'
+                }
+                'SqlServer2008' { Start-Service 'MSSQL$SQL2017' }
+                'SqlServer2012' { Start-Service 'MSSQL$SQL2017' }
```

Comment:
The `Visual Studio 2017` image does not have SQL 2008 and SQL 2012, it has [2014, 2016 and 2017](https://www.appveyor.com/docs/build-environment/#databases). So I have picked one of them.